packages/calypso-e2e/src/totp-client.ts

```diff
@@ -0,0 +1,24 @@
+import totp from 'totp-generator';
+
+/**
+ *
+ */
+export class TOTPClient {
+	private secret: string;
+
+	/**
+	 *
+	 * @param secret
+	 */
+	constructor( secret: string ) {
+		this.secret = secret;
+	}
+
+	/**
+	 *
+	 * @returns {string} TOTP code valid for the epoch for which it was generated.
+	 */
+	getToken(): string {
+		return totp( this.secret ).toString();
+	}
+}
```

Comment:
Totally optional, but do you see this functionality growing significantly?

If this class just will be used to generate a token, what do you think about making just a globally exported library function that takes the secret at the first parameter? Again, a totally optional style thing, just wanted to throw it out there!

Reply:
Functionally I do not see this class growing at all.

With that said, I think it might be worth keeping it as a class to draw parallel to the `EmailClient` which is in its own class. I realize the scope of its actions is vastly different, but conceptually it makes more sense (to me) that TOTP is implemented as a separate client that needs to be invoked.